docs/content/topics/project-developers/events.md

```diff
@@ -9,4 +9,229 @@ aliases:
   - /topics/project-developers/events.md
 ---
 
-TODO
\ No newline at end of file
+# Brigade Events
+
+Events are the lingua franca in Brigade: [Gateways] emit events and [Project]
+developers write logic to handle them.
+
+In this document, we'll look at:
+
+  * The [structure of an event]
+  * How to [configure event subscriptions] on a project
+  * How to [handle events] in a Brigade script.
+
+[Gateways]: /topics/operators/gateways
+[Project]: /topics/project-developers/projects
+[structure of an event]: #event-structure
+[configure event subscriptions]: #event-subscriptions
+[handle events]: #handling-events
+
+## Event Structure
+
+A Brigade event is defined primarily by its source and type values. These two
+must be provided every time an event is emitted into Brigade. However, there
+are plenty of other fields that event creators can harness to add additional
+context and information for utilization in handlers.
+
+The full list of fields on an Event is:
+
+  * [Source](#source)
+  * [Type](#type)
+  * [Payload](#payload)
+  * [ProjectID](#project-id)
+  * [Qualifiers](#qualifiers)
+  * [Labels](#labels)
+  * [Short Title](#short-title)
+  * [Long Title](#long-title)
+  * [Source State](#source-state)
+  * [Summary](#summary)
+
+Each field is reviewed in depth below.
+
+### Source
+
+An event's Source can be thought of as its domain. In fact, the rule of thumb
+to avoid clashes is to use a URI you control. This means leading with one's own
+domain or the URL for something else one owns, like the URL for a GitHub repo.
+As an example, events emitted from Brigade's [GitHub Gateway] use a source
+value of `brigade.sh/github`.
+
+### Type
+
+The Type of an event is used to specify increased granularity of the event's
+role/purpose as it relates to the source. For example, an event emitted by
+the GitHub Gateway with Type `push` signifies that a push event has occurred
+on a particular GitHub repo.
+
+### Payload
+
+An event Payload can be used to send free-form input that may be utilized when
+handling the event. One example would be sending the original GitHub push event
+payload on the corresponding GitHub gateway event, so that these additional
+details may be parsed in the Brigade script.
+
+### Project ID
+
+Although not normally used by general-purpose gateways, a Project ID value may
+be set on an event. In such cases, the event will _only_ be eligible for
+receipt by the project indicated by this fields' value.
+
+### Qualifiers
+
+For a project to receive a particular Eeent, the qualifiers on the project's
+Eeent subscription must exactly match the qualifiers on the event itself (in
+addition to matching Source and Type). For example, a project might supplement
+its `brigade.sh/github` event subscription with a qualifier of
+`repo: brigadecore/brigade`. Only events exactly matching this qualifier would
+be received by the project, i.e., only GitHub events on the brigadecore/brigade
+repository.
+
```

Comment:
I think you accurately stated how qualifiers work, but I'm worried the example might resemble regular label usage.

Do you think it would be useful to introduce labels _first_ as something that _optionally_ narrows a subscription and _then_ describe qualifiers as "like labels" except _required_ in order to match an event?

Reply:
Do you suppose that for qualifiers and labels (and I guess source and type for that matter, as well) we should explicitly state that gateways are expected to document what kind of metadata they attach to their events?

Reply:
> Do you suppose that for qualifiers and labels (and I guess source and type for that matter, as well) we should explicitly state that gateways are expected to document what kind of metadata they attach to their events?

I think this might still be outstanding.